### DISCUSSION TITLE
Is it possible to use http.Response.content as iterator in Addon

### DISCUSSION THREAD

**Discussion by melehin** [mitmproxy/mitmproxy#5066] [Q&A] [answered]
Hi!

I made a simple **Addon** and it works fine. This **Addon** forwards a request to a http endpoint which is a simple aiohttp app which uses **Telegram protocol as transport layer**.
```python
def request(flow: http.HTTPFlow) -> None:
    r = requests.post("http://localhost:12345/", data=export.raw_request(flow), headers={'URL': flow.request.url}, timeout=3600)
    flow.response = http.Response.make(
        r.status_code,
        r.content,
        {**r.headers},
    )
```

my idea is to use **r.iter_content** instead of **r.content** because for big files my addon requires a lot of RAM.
Is there a way to use **iterator** instead of **bytes** or **str**?
```python
def request(flow: http.HTTPFlow) -> None:
    r = requests.post(.., stream=True)
    flow.response = http.Response.make(
            r.status_code,
            r.iter_content,
            {**r.headers},
        )
```

Or should i use **--mode reverse:http://localhost:12345/** and adopt my http endpoint to use CONNECT method?

**Comment by mhils** [comment, marked as answer]
> Is there a way to use iterator instead of bytes or str?

Not right now, and I don't think this is feasible to implement for us in mitmproxy. The iterator would block when reading additional bytes, and we cannot block in the proxy core or else everything would come to a halt.

**Reply by melehin** [reply]
Thanks!
